### PR TITLE
fix: getsessionid updating the sessionstart status

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Notion ticket

https://linear.app/rudderstack/issue/SDK-574/fix-getsessionid-updating-the-sessionstart-status

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-js-plugins/.size-limit.js [hunk 1]
```diff
@@ -13,6 +13,6 @@ module.exports = [
     name: 'Remote Module Federated Plugins - CDN',
     path: 'dist/cdn/modern/plugins/rsa-plugins-*.js',
     gzip: true,
-    limit: '15 KiB',
+    limit: '19 KiB',
   },
```
Comment (bardisg) [inline packages/analytics-js-plugins/.size-limit.js:17]: Delete the dist folder before the precommit checks

File: packages/analytics-js/src/components/core/Analytics.ts [hunk 2]
```diff
@@ -689,8 +689,8 @@ class Analytics implements IAnalytics {
 
   // eslint-disable-next-line class-methods-use-this
   getSessionId(): Nullable<number> {
-    this.userSessionManager?.refreshSession();
-    return state.session.sessionInfo.value?.id ?? null;
+    const sessionId = this.userSessionManager?.getSessionId();
```
Comment (saikumarrs) [inline packages/analytics-js/src/components/core/Analytics.ts:692]: This seems incorrect. If the session expires, the caller will not get the refreshed value.

Comment (saikumarrs) [reply packages/analytics-js/src/components/core/Analytics.ts:692]: We need to have a different strategy for marking the session start status.

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: packages/analytics-js/src/components/core/Analytics.ts (hunk 2 above)
Comment (saikumarrs) [reply packages/analytics-js/src/components/core/Analytics.ts:692]: I am creating another task for changing the strategy for marking the session start status.

**Review by saikumarrs — APPROVED** [review]

File: packages/analytics-js/src/components/core/Analytics.ts (hunk 2 above)
Comment (saikumarrs) [reply packages/analytics-js/src/components/core/Analytics.ts:692]: https://linear.app/rudderstack/issue/SDK-576/change-the-strategy-to-denote-the-mark-of-session-start

File: packages/analytics-js-plugins/.size-limit.js (hunk 1 above)
Comment (MoumitaM) [reply packages/analytics-js-plugins/.size-limit.js]: Done

File: packages/analytics-js-plugins/.size-limit.js [hunk 3]
```diff
@@ -13,6 +13,6 @@ module.exports = [
     name: 'Remote Module Federated Plugins - CDN',
     path: 'dist/cdn/modern/plugins/rsa-plugins-*.js',
     gzip: true,
-    limit: '15 KiB',
+    limit: '19 KiB',
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply packages/analytics-js-plugins/.size-limit.js]: Reverted the change

**Pull request merged**